Fixed `rudderboat`/`throttleboat` prop anims

ohlidalp opened the pull request:
Fixes #3174

ohlidalp committed:
:bug: Fixed `rudderboat`/`throttleboat` prop anims

Known issue: throttleboat + autoanimate doesn't seem to work at all (i.e. Squalo boat).

CuriousMike56 commented:
@ohlidalp Squalo props are actually functional, the problem is that they stop moving at 55+% throttle (shown only in T-menu, HUD throttle visualization got removed at some point). This is easily visible at 10% physics speed:

https://github.com/user-attachments/assets/2cd833d2-cbd5-4793-a442-44156bdce942

ohlidalp commented:
I have a hunch they don't really stop spinning, just start rotating by exactly 180° each frame, creating illusion of no motion.

ohlidalp committed:
:bug: Fixed prop anim limiters not accounting time right

This was broken by myself all the way back in 7767982a8a282f4d305969e4b10a8d0047dcf67b (8 years ago) when I separated prop animation code from 'animators' code.
Animators ran on physics timestep (2khz) and prop animations newly ran on FPS rate. I accounted for this in the offseting/rotation code but not in the limiters code.
